### DISCUSSION TITLE
All of a sudden I am getting that 'spillover' isn't a key word

### DISCUSSION THREAD

**Discussion by madinagy** [whitews/FlowKit#214] [General]
Hi! I've been working with the same file and from time to time I get that 'spillover' does not work and I get an error. I'm working with version 1.1.2. Any suggestions on how to fix this? Thanks!

**Comment by whitews** [comment]
Hi @madinagy,

It's difficult to say without seeing the code and the error (preferably with the stack trace). Any keywords in an FCS file's metadata should be available via the `Sample.metadata` attribute, there's no filtering applied. Always possible to have a bug, so would be interested in seeing the stack trace.

Kind regards,
Scott

**Comment by madinagy** [comment]
Hi Scott!
Thanks for the quick response. I found out that if you are using FlowJo to
first export the fcs file and select all compensated parameters the
['spillover'] keyword is no longer available. I'm not totally sure why but
thought that was pretty interesting.

**Comment by whitews** [comment]
Interesting, maybe they were thinking if the compensated events are being exported then the matrix is no longer necessary. Just a guess, but I'm glad you figured it out.

**Comment by madinagy** [comment]
Quick question for you: How would you recommend saving a fcs file with the
new compensation value changes I've made? Currently, I am saving it as
 'sample.export(f'comps{timepoint}_{sets}.fcs', source= 'comps',
subsample=False)' but the graphs with my new compensation look nearly
identical to the old ones. I am thinking about changing it to xform? What
do you think?

Thanks!
Madi

**Comment by whitews** [comment]
Hi Madi,

It should be `source='comp'` not 'comps'. I think that's why you aren't seeing a difference.

-Scott

**Comment by madinagy** [comment]
Thank you so much! I have been using
https://flowkit.readthedocs.io/en/latest/notebooks/flowkit-tutorial-part01-sample-class.html
as a resource as well which makes sense as to why I am not seeing the
results! On this website it uses 'comp'.

Thanks!
Madi

**Comment by madinagy** [comment]
Sorry I misread your email. I am using 'comp', but for some reason I use
        sample.apply_compensation(sample.metadata['spillover'])

        # Remove or comment out the following line to avoid exporting files
        sample.export(f'comps{timepoint}_{sets}.fcs', source= 'comp',
subsample=False)
and when I put the compensated file back into FlowJo it is exactly the
same. Do I need to move apply_compensation elsewhere?

thanks!
Madi

**Comment by whitews** [comment]
What do you see if you load the comp sample back info FlowKit as a new Sample? Is it changed then?

**Comment by madinagy** [comment]
[image: FlowKit Comp.png][image: Original.png][image: bokeh_plot (28).png]

So the first two images are the original image and the compensated values.
The last image is using a bokeh plot. What do you imagine is going on here?

**Comment by whitews** [comment]
Sorry, the images didn't come through. I think you'll have to add them on the GitHub page & not through email.

**Comment by madinagy** [comment]
<img width="388" alt="FlowKit Comp" src="https://github.com/user-attachments/assets/f062a489-11be-4013-9806-27b80196f0ce">
<img width="390" alt="Original" src="https://github.com/user-attachments/assets/d00076a8-dd29-4f3b-be8a-fc0f15cd673d">
<img width="659" alt="Screenshot 2024-08-14 at 2 16 31 PM" src="https://github.com/user-attachments/assets/74158e15-db4c-48ab-8081-9ecc17440f6f">

**Comment by whitews** [comment]
The FlowJo plots appear to have the default biex transform applied but the FlowKit Bokeh plot looks untransformed. I suspect the original in FlowJo is pulling the spillover and automatically applying it. You could also check the compensation matrix to make sure it isn't a simple identity matrix.

**Comment by whitews** [comment]
Looking again, the original in FlowJo is showing it compensated, you can tell by the 'Comp-' prefix in the channel labels. In the comp'd export FlowJo is showing the source event data in the file (which was, of course, compensated). So it would be expected that they would be the same, they are both compensated. Hope that helps.

**Comment by madinagy** [comment]
Thank you so much for the clarification and help! I appreciate the quick
response and helpful feedback!
